src/lib/evil/evil_unistd.h

```diff
@@ -28,6 +28,11 @@ EVIL_API int execvp(const char *file, char *const argv[]);
 #define W_OK    2       /* Test for write permission.  */
 #define X_OK    0       /* execute permission, originally '1', just a bypass here*/
 #define F_OK    0       /* Test for existence.  */
+
+#define STDIN_FILENO 0
+#define STDOUT_FILENO 1
+#define STDERR_FILENO 2
```

Comment:
What are they for?

Reply:
> What are those for?

These are standard "file" identifiers for stdin/out/err, just as it happens when you use `command 2>/dev/null` to redirect errors to `/dev/null`.

Currently it is used in these places:

```
$ git grep STDIN_FILENO src
src/bin/edje/edje_player.c:        flags = fcntl(STDIN_FILENO, F_GETFL, 0);
src/bin/edje/edje_player.c:        if (fcntl(STDIN_FILENO, F_SETFL, flags) < 0)
src/bin/edje/edje_player.c:        ecore_main_fd_handler_add(STDIN_FILENO, ECORE_FD_READ | ECORE_FD_ERROR,
src/bin/ethumb_client/ethumbd_slave.c:      STDIN_FILENO, ECORE_FD_READ | ECORE_FD_ERROR,
src/examples/ecore/ecore_con_client_example.c:   ecore_main_fd_handler_add(STDIN_FILENO, ECORE_FD_READ, _on_stdin, NULL, NULL, NULL);
src/examples/ecore/ecore_evas_basics_example.c:   ecore_main_fd_handler_add(STDIN_FILENO, ECORE_FD_READ, _stdin_cb, NULL, NULL, NULL);
src/examples/ecore/ecore_evas_ews_example.c:   ecore_main_fd_handler_add(STDIN_FILENO,
src/examples/ecore/ecore_exe_example_child.c:   ecore_main_fd_handler_add(STDIN_FILENO,
src/examples/ecore/ecore_fd_handler_example.c:   ctxt.handler = ecore_main_fd_handler_add(STDIN_FILENO,
src/examples/ecore/ecore_ipc_client_example.c:   ecore_main_fd_handler_add(STDIN_FILENO, ECORE_FD_READ, _on_stdin, NULL, NULL, NULL);
src/lib/ecore/Ecore_Common.h: * ecore_main_fd_handler_add(STDIN_FILENO, ECORE_FD_READ, _my_cb_func, NULL, NULL, NULL);
src/lib/ecore/ecore_exe_posix.c:                   E_NO_ERRNO(result, close(STDIN_FILENO), ok);
src/lib/ecore/ecore_exe_posix.c:                   E_NO_ERRNO(result, dup2(devnull, STDIN_FILENO), ok);
src/lib/ecore/ecore_exe_posix.c:                   E_NO_ERRNO(result, close(STDIN_FILENO), ok);
src/lib/ecore/ecore_exe_posix.c:                   E_NO_ERRNO(result, dup2(writePipe[0], STDIN_FILENO), ok);
src/lib/ecore/efl_exe.c:     close(STDIN_FILENO);
src/lib/ecore/efl_exe.c:        dup2(devnull, STDIN_FILENO);
src/lib/ecore/efl_exe.c:        dup2(pipe_stdin[0], STDIN_FILENO);
src/lib/ecore/efl_io_stderr.c:   if (fd < 0) efl_loop_fd_set(o, STDIN_FILENO);
src/lib/ecore/efl_io_stdin.c:   if (fd < 0) efl_loop_fd_set(o, STDIN_FILENO);
src/lib/ecore_drm/ecore_drm_tty.c:             dev->tty.fd = dup(STDIN_FILENO);
```